drivers/SmartThings/matter-lock/src/new-matter-lock/init.lua

```diff
@@ -0,0 +1,1800 @@
+  device:send(DoorLock.server.commands.ClearCredential(device, ep, credential))
+end
+
+----------------------------
+-- Delete All Credentials --
+----------------------------
+local function handle_delete_all_credentials(driver, device, command)
+  log.info_with({hub_logs=true}, string.format("!!!!!!!!!!!!!!! handle_delete_all_credentials !!!!!!!!!!!!!")) -- needs to remove
+
+  -- Get parameters
+  local cmdName = "deleteAllCredentials"
+  local credential = {
+    credential_type = DoorLock.types.DlCredentialType.PIN,
+    credential_index = ALL_INDEX,
+  }
+
+  -- Check busy state
+  local busy = device:get_field(lock_utils.BUSY_STATE)
+  if busy == true then
+    local result = {
+      commandName = cmdName,
+      statusCode = "busy"
+    }
+    local event = capabilities.lockCredentials.commandResult(
+      result,
+      {
+        state_change = true,
+        visibility = {displayed = false}
+      }
+    )
+    device:emit_event(event)
+    return
+  end
+
+  -- Save values to field
+  device:set_field(lock_utils.BUSY_STATE, true, {persist = true})
+  device:set_field(lock_utils.COMMAND_NAME, cmdName, {persist = true})
+  device:set_field(lock_utils.CRED_INDEX, ALL_INDEX, {persist = true})
+
+  -- needs to remove logs
+  log.info_with({hub_logs=true}, string.format("commandName: %s", cmdName))
+  log.info_with({hub_logs=true}, string.format("credentialIndex: %s", ALL_INDEX))
+
+  -- Send command
+  local ep = device:component_to_endpoint(command.component)
+  device:send(DoorLock.server.commands.ClearUser(device, ep, credential))
+end
+
+-------------------------------
+-- Clear Credential Response --
+-------------------------------
+local function clear_credential_response_handler(driver, device, ib, response)
+  log.info_with({hub_logs=true}, string.format("!!!!!!!!!!!!!!! clear_credential_response_handler !!!!!!!!!!!!!")) -- needs to remove
+
+  -- Get result
+  local cmdName = device:get_field(lock_utils.COMMAND_NAME)
+  local credIdx = device:get_field(lock_utils.CRED_INDEX)
+  local status = "success"
+  if ib.status == DoorLock.types.DlStatus.FAILURE then
+    status = "failure"
+  elseif ib.status == DoorLock.types.DlStatus.INVALID_FIELD then
+    status = "invalidCommand"
+  end
+
+  -- Delete User in table
+  local userIdx = 0
+  if status == "success" then
+    userIdx = delete_credential_from_table(device, credIdx)
+  end
+  
+  -- Update commandResult
+  local result = {
+    commandName = cmdName,
+    userIndex = userIdx,
+    credentialIndex = credIdx,
+    statusCode = status
+  }
+  local event = capabilities.lockCredentials.commandResult(
+    result,
+    {
+      state_change = true,
+      visibility = {displayed = false}
+    }
+  )
+  device:emit_event(event)
+  device:set_field(lock_utils.BUSY_STATE, false, {persist = true})
+end
+
+---------------------------
+-- Set Week Day Schedule --
+---------------------------
+local function handle_set_week_day_schedule(driver, device, command)
```

Comment:
@HunsupJung what is the expected behavior if the lock does not currently know the time? For example, we do not support time sync at the moment and would require the time to be synced to the device from elsewhere. Is there a way to check if the device has the time, and if not, then we would disable scheduling?